### DISCUSSION TITLE
Not able to see OnPrompt is being called.

### DISCUSSION THREAD

**Discussion by msita143** [SupremeTechnopriest/react-idle-timer#278] [Q&A] [answered]
I was just testing on how this works, and trying to understand and found the onPrompt function is not being called.
Here is my simple code:

`
import React, { useState, useEffect } from "react";
import { useIdleTimer } from "react-idle-timer";

export default function App() {

  const onPrompt = () => {
    console.log("Now I am ON PROMOT...");
  };
  const onIdle = () => {
    console.log("Now I am idle... : ");
  };

  const onActive = (event) => {
    console.log("Now I am Active...", event);
  };

  const onAction = (event) => {
    console.log("Now I am taking action...");
  };

  const { getRemaningTime, isPrompted, getTabId, activate } = useIdleTimer({
    timeout: 1000 * 20, //20 secs
    promptTimeout: 0, //10sec
    onPrompt,
    onIdle,
    onAction,
    onActive,
    //events: ["mousemove", "mousedown"],
    //eventsThrottle: 200,
    startOnMount: true
    //element: document
  });

  return (
    <div>

        <h1>Timeout: {timeout}ms</h1>
        <h1>Time Remaining: {remaining}</h1>
        <h1>Time Elapsed: {elapsed}</h1>
        <h1>Last Active: {format(lastActive, "MM-dd-yyyy HH:MM:ss.SSS")}</h1>
    </div>
  );
}
`

**Comment by SupremeTechnopriest** [comment]
Looks like your prompt timeout is 0. That needs to be a positive, non zero number for the onPrompt function to be called.

**Reply by msita143** [reply]
I did try with 10 sec and it still not being called

**Reply by msita143** [reply]
<img width="1878" alt="image" src="https://user-images.githubusercontent.com/17054088/187127787-c0990f2c-8150-488e-ab43-dc7fef172c2b.png">

**Reply by msita143** [reply]
it is printing idle but not prompt stuff

**Comment by msita143** [comment]
I was using old 4.o. IdleTimer and came to know this new 5.0 one.
Now this onPrompt is not very clear. In 4.0 we have our Modal which pops up 1 min before idle is going to happen.
but now not sure how do I do it here as onPrompt is not working

**Comment by SupremeTechnopriest** [comment]
Can you link me to that code sandbox.

**Comment by SupremeTechnopriest** [comment, marked as answer]
Looks like that sandbox is using v4.  Prompts only work with v5.

**Reply by msita143** [reply]
Oh yes , sorry my bad, I did not check the version.
Yes it works.
Thanks for all the help

**Reply by SupremeTechnopriest** [reply]
No problem. Can you mark this as answered?

**Reply by SupremeTechnopriest** [reply]
Nevermind I got it!
